daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/Converter.scala

```diff
@@ -219,24 +219,17 @@ object Converter {
         seeding = InitialSeeding.NoSeed,
         Set.empty,
       )
-    @tailrec
-    def iter(): Either[String, (SValue, SValue)] = {
-      if (machine.isFinal) {
```

Comment:
Why do we unconditionally run the machine now?

Reply:
I don't believe anything has changed.
It was being unconditionally run before. `step`ing until `isFinal`

Reply:
That makes sense. Thanks.